### ISSUE TITLE
Rotary encoder dial on back

### ISSUE THREAD

**Issue by MatthewPageUK** [MatthewPageUK/pico_desktop_clock#18]
Have a rotary dial on the back which can be used to set the time and maybe scroll through other features or settings.

Looking at using this 
https://thepihut.com/products/ano-directional-navigation-and-scroll-wheel-rotary-encoder

![Image](https://user-images.githubusercontent.com/46349796/218342377-b8830ce8-df00-4162-bb71-347c56a9475a.png)

Fit is going to be tight and may need re-wiring the Pico from below.

**Comment by MatthewPageUK** [comment]
![image](https://user-images.githubusercontent.com/46349796/218342461-57293237-4b2c-4af6-b42f-5ac2004778dc.png)

**Comment by samveen** [comment]
A couple of observations about options:

Option 1 - use  an El Cheapo encoder (following the usage pattern for El Cheapo children's watches)  that just senses rotation:
   - Long press of one Button1/Button2 enters modification mode
   - In modification mode, quick press of buttons cycle through modifiable selections 
   - Rotation always increments, and loops back to minimum at maximum
   - Pros: Minimal/cheap hardware, just one extra hole in the case
   - Cons: Lots of hoops to jump though in software, the repetitions in this usage pattern get old really quick.
 
Option 2 - Bells and whistles encoder
   - Ease of use
   - Non-repetitive usage pattern
   -  Pros: Quicker usage
   - Cons: Not as easily accessible hardware, lots of extra wiring and design considerations (made easier by https://thepihut.com/products/adafruit-ano-rotary-navigation-encoder-breakout-pcb).

**Comment by MatthewPageUK** [comment]
I agree with all your points there. I really like the Bells and Whistles encoder, feels nice, same diameter as the LCD, but it is a bit of work to redesign the case etc. 

Good point about it not being easily accessible hardware. Should try to keep it to simple parts that could be swapped with something similar if you can't get the exact one.

Most of the other encoders are very big, but I have just found **mouse wheel encoders**

![image](https://user-images.githubusercontent.com/46349796/218445168-c971938d-56d9-4654-81a8-cceda122cfc9.png)

https://uk.rs-online.com/web/p/mechanical-rotary-encoders/7295536?cm_mmc=UK-PLA-DS3A-_-google-_-CSS_UK_EN_Passive_Components_Whoop-_-Mechanical+Rotary+Encoders_Whoop_OMNISerpNov-_-7295536&matchtype=&pla-341413212373&cq_src=google_ads&cq_cmp=9771206638&cq_term=&cq_plac=&cq_net=g&cq_plt=gp&gclsrc=ds&gclsrc=ds

Small, easily available and could be swapped out for a different encoder if you can't get the exact model.
